src/features/watchLater/watchLater-slice.jsx

```diff
@@ -0,0 +1,79 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import axios from "axios";
+
+const initialState = {
+  watchLater: [],
+  status: null,
+};
+
+let encodedToken = "";
+
+export const addToWatchLater = createAsyncThunk(
+  "watchLater/addToWatchLater",
+  async (video) => {
+    try {
+      encodedToken = localStorage.getItem("token");
+      const response = await axios.post(
+        "/api/user/watchlater",
+        { video },
+        {
+          headers: {
+            authorization: encodedToken,
```

Comment:
suggestion: instead of writing an authorization header, you can set the default header with token details.

Reply:
ok